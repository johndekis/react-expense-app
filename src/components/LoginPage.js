import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
        <h1 className="box-layout__title">Expenster</h1>
        <p>Track your expenses and eliminate them</p>
        <button className="button login" onClick={startLogin}>
            <span className="gcon"></span>
            <span className="button-text">Login with Google</span>
        </button>
        </div>        
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);