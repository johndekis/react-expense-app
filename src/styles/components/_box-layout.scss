.box-layout {
    align-items: center;
    background: url('/images/bg.jpg');
    background-size: cover;
    display: flex;
    justify-content: center;
    height: 100vh;
    width: 100vw;
}

.box-layout p {
    color: #333;
}

.box-layout__box {
    background: fade-out(white, .15);
    border-radius: 3px;
    padding: $l-size $m-size;
    text-align: center;
    width: 25rem;

}

.box-layout__title {
    font-family: 'Rajdhani', sans-serif;
    margin: 0 0 $m-size 0;
    line-height: 1;
    color: #333;
}