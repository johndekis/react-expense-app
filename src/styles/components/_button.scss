.button {
    background-color: $bright-green;
    border: none;
    border-radius: .4em;
    color: white;
    display: inline-block; 
    font-size: $font-size-small;
    font-weight: 300;
    line-height: 1;   
    padding: $s-size;
    text-decoration: none;
    &:hover {
        color: $bright-green;
        border: 1px solid $bright-green;
        background-color: #444;
    }
}

.button__link {
    background: none;
    &:hover {
        background: #222;
    }
}


.button__secondary {
    background: #888;
}

.gcon {
    display: inline-block;
    background: url('/images/google.svg');
    height: 1.5em;
    width: 1.5em;
    margin-right: .5em;
}

.login {
    background-color: #333;
    margin: 0 auto;
    width: 12em;
    display: flex;
    justify-content: space-around;
    align-items: center;
}