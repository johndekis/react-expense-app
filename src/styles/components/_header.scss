.header {
    background: $dark-grey;
}

.header__content {
    align-items: center;
    display: flex;
    justify-content: space-between;
}

.header__title {
   color: white;
   font-family: 'Rajdhani', sans-serif;
   text-decoration: none;
   h1 {
       margin: 0;
   } 
}