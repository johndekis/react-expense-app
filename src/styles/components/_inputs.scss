.text-input {
   border: none;
   height: 50px;
   font-size: $font-size-large;
   padding: $s-size;
   color: white;
   background: #333;
   &:focus {
       outline: 2px solid $bright-green;
       color: $bright-green;
   }
}

.select {
    @extend .text-input;
    
}

.textarea {
    @extend .text-input;
    height: 10rem;
}

::placeholder { /* Chrome, Firefox, Opera, Safari 10.1+ */
    color: white;
    opacity: 1; 
  }
 