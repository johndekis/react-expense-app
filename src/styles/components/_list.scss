.list-header {
    background: $dark-grey;
    border: 1px solid lighten($grey, 7%);
    color: whitesmoke;
    display: flex;
    justify-content: space-between;
    padding: $s-size $m-size;
}

.list-body {
   margin-bottom: $m-size;
   @media (min-width: $desktop-breakpoint) {
       margin-bottom: $l-size;
   } 
}

.list-item {
    background: $grey;
    border: 1px solid lighten($grey, 7%);
    border-top: none;
    color: white;
    display: flex;
    flex-direction: column;
    padding: $s-size;
    text-decoration: none;
    transition: background .3s ease;
    &:hover {
        background: $dark-grey;
        border: none;
        text-shadow: 0.01em 0.01em 0.09em white;
        .list-item__data {
            color: $bright-green;
            
            text-shadow: 0.01em 0.01em 0.09em;
        }
    }
    @media (min-width: $desktop-breakpoint) {
        align-items: center;
        flex-direction: row;
        justify-content: space-between;
        padding: $m-size;
    }
}

.list-item__message {
    align-items: center;
    color: $light-grey;
    justify-content: center;
    padding: $m-size;
    &:hover {
        background: none;
    }
}

.list-item__title {
    margin: 0;
    word-break: break-all;
    font-size: 3rem;
}

.list-item__sub-title {
    color: $off-white;
    font-size: $font-size-small;
}

.list-item__sub-title::before {
    content: "\2022";  
    color: $bright-green; 
    font-weight: bold;
    display: inline-block; 
    width: 1em; 
}

.list-item__data {
    margin: $s-size 0 0 0;
   
    @media (min-width: $desktop-breakpoint) {
        margin: 0;
        padding-left: $s-size;
    }
}

.note {
    margin: 0 0 0 2em;
    font-size: .7em;
    color: $off-white;
}