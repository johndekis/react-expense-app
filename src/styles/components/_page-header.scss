.page-header {
    background: #444;
    color: whitesmoke;
    margin-bottom: $l-size;
    padding: $l-size 0;
}

.page-header__actions {
    margin-top: $m-size;
}

.page-header__title {
    font-weight: 300;
    margin: 0;
    span {
        color: $bright-green;
        font-weight: 700;
    }

}
