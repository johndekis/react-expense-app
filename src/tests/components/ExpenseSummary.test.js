import React from 'react';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import { shallow } from 'enzyme';

test('should show 0 expenses and $0 when no expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={0} expensesTotal={0} />);
    expect(wrapper).toMatchSnapshot();
});

test('should show number and total of visible expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={10} expensesTotal={27568796054857} />);
    expect(wrapper).toMatchSnapshot();
});