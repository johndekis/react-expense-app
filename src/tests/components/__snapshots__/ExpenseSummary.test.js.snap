// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show 0 expenses and $0 when no expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        0
      </span>
       
      expenses
       totalling 
      <span>
        $0.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should show number and total of visible expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        10
      </span>
       
      expenses
       totalling 
      <span>
        $275,687,960,548.57
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
